pkg/k8s/service_cache.go

```diff
@@ -129,26 +130,26 @@ func (s *ServiceCache) GetServiceFrontendIP(svcID ServiceID, svcType loadbalance
 	s.mutex.RLock()
 	defer s.mutex.RUnlock()
 	svc := s.services[svcID]
-	if svc == nil || svc.Type != svcType {
+	if svc == nil || svc.Type != svcType || len(svc.FrontendIPs) == 0 {
 		return nil
 	}
 
-	return svc.FrontendIP
+	return svc.FrontendIPs[0]
 }
 
-// GetServiceAddrWithPortsAndType returns a slice of all the L3n4Addr that are backing the
+// GetServiceAddrsWithType returns a slice of all the L3n4Addr that are backing the
 // given Service ID with given type.
 // Note: The returned IPs are with External scope.
 func (s *ServiceCache) GetServiceAddrsWithType(svcID ServiceID, svcType loadbalancer.SVCType) map[loadbalancer.FEPortName]*loadbalancer.L3n4Addr {
 	s.mutex.RLock()
 	defer s.mutex.RUnlock()
 	svc := s.services[svcID]
-	if svc == nil || svc.Type != svcType {
+	if svc == nil || svc.Type != svcType || len(svc.FrontendIPs) == 0 {
 		return nil
 	}
 	addrsByPort := make(map[loadbalancer.FEPortName]*loadbalancer.L3n4Addr)
 	for pName, l4Addr := range svc.Ports {
-		addrsByPort[pName] = loadbalancer.NewL3n4Addr(l4Addr.Protocol, svc.FrontendIP,
+		addrsByPort[pName] = loadbalancer.NewL3n4Addr(l4Addr.Protocol, svc.FrontendIPs[0],
```

Comment:
same here? ... I'm starting to see a pattern but I still don't understand the why :slightly_smiling_face:

Reply:
Here we should not pick a single IP address, rather we should return all the valid frontend IPs associated with the service. I will update the PR.